src/twisted/internet/_idna.py

```diff
@@ -28,10 +28,10 @@ def _idnaBytes(text):
     except ImportError:
         return text.encode("idna")
     else:
-        return idna.encode(text)
+        return idna.encode(text)  # type: ignore[no-any-return]
```

Comment:
this has a type annotation, so I think the twisted mypy setup might be wrong

https://github.com/kjd/idna/blob/master/idna/core.py#L354

I think the all_deps_nonplatform extra will need to be installed in the mypy toxenv

Reply:
@wsanchez ^

Reply:
The last release of `idna` was on Jan 4 this year, while `py.typed` was added on Jan 19. So unless you'd pin a pre-release snapshot, mypy won't see any type annotations.

Reply:
I remain confused about Twisted's build setup, but if `all_deps_nonplatform` installs dependencies, we do want it in the `mypy` Tox environment, so that it can use whatever annotations are (or become) available there.